packages/application/src/lab.ts

```diff
@@ -31,16 +35,8 @@ export class JupyterLab extends JupyterFrontEnd<ILabShell> {
         })
     });
 
-    // Create an IInfo dictionary from the options to override the defaults.
-    const info = Object.keys(JupyterLab.defaultInfo).reduce((acc, val) => {
-      if (val in options) {
-        (acc as any)[val] = JSON.parse(JSON.stringify((options as any)[val]));
-      }
-      return acc;
-    }, {} as Partial<JupyterLab.IInfo>);
-
     // Populate application info.
-    this._info = { ...JupyterLab.defaultInfo, ...info };
+    this._info = new JupyterLab.Info(options);
```

Comment:
Note to self: this will prevent writing to some attributes, but it's ok because it is private